File: src/main/java/com/example/chatService/redis/WsTokenService.java
package com.example.chatService.redis;

import com.example.chatService.dto.WsTokenResponse;
import com.example.chatService.security.CookieUtil;
import com.example.chatService.security.JwtTokenProvider;
import jakarta.servlet.http.HttpServletRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.time.Duration;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class WsTokenService {

    private final RedisTemplate<String, Object> redisTemplate;
    private static final String KEY_PREFIX = "ws_token:";
    private static final int TTL_SECONDS = 120;

    public String createTokenForUser(String userId) {
        String token = UUID.randomUUID().toString();
        String key = KEY_PREFIX + token;
        redisTemplate.opsForValue().set(key, userId, Duration.ofSeconds(TTL_SECONDS));
        return token;
    }

    public String consumeToken(String token) {
        String key = KEY_PREFIX + token;
        String userId = (String) redisTemplate.opsForValue().get(key);
        if (userId != null) {
            // 1회용: 즉시 삭제
            redisTemplate.delete(key);
        }
        return userId;
    }

    public String peekToken(String token) {
        return (String) redisTemplate.opsForValue().get(KEY_PREFIX + token);
    }
}

File: src/main/java/com/example/chatService/redis/WsTokenService.kt
package com.example.chatService.redis

import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Service
import java.time.Duration
import java.util.UUID

@Service
class WsTokenService(
    private val redisTemplate: RedisTemplate<String, Any>,
) {
    fun createTokenForUser(userId: String): String {
        val token = UUID.randomUUID().toString()
        val key = "$KEY_PREFIX$token"
        redisTemplate.opsForValue().set(key, userId, Duration.ofSeconds(TTL_SECONDS.toLong()))
        return token
    }

    fun consumeToken(token: String): String? {
        val key = "$KEY_PREFIX$token"
        val userId = redisTemplate.opsForValue().get(key) as? String
        if (userId != null) {
            redisTemplate.delete(key)
        }
        return userId
    }

    fun peekToken(token: String): String? = redisTemplate.opsForValue().get("$KEY_PREFIX$token") as? String

    companion object {
        private const val KEY_PREFIX = "ws_token:"
        private const val TTL_SECONDS = 120
    }
}
